src/popup/modules/QrLib/retryPromise.js

```diff
@@ -0,0 +1,15 @@
+/**
+ * Try fn when the return promise is rejected
+ * @param {Function} fn function to retry
+ * @param {number} ms retry delay in ms
+ * @returns {Promise}
+ */
+const retryPromise = (fn, ms) => new Promise(resolve => {
```

Comment:
Wow, pretty clever stuff here and functional programming as it should be. ☺️

Just some minor notes:
* do not abbreviate the var names so much, rather call it "delay"/"time" or so, same with "function". We have enough space here. :wink: 
* you can (and may :wink:) use the usual `function(` syntax here instead of arrow functions and a variable assignment. Just so it looks less creepy to ones knew to these stuff… :wink: 
* everything else, great! Maybe move it to the `commons` dir though, as it is not limited to the popup functionality…
   and, yeash, until now I did not use default exports. Maybe just put it into the "HelperFunctions.js" file there – or how it is called – …

Reply:
Thank you, i'm a big fan of simplicity and composability.
I'll address these points and make declarations more descriptive.